### PR TITLE
feat: align x-axis ticks with data

### PR DESCRIPTION
**Pull request by magnew**
Closes: #3621 

### Description:

The goal of this is to make data line up with the axis labels and ticks better when looking at WEEK timeframe data. 

To do this, I made the week-timeframe use a categorical axis, which gives much more control over axis styles and such. The downside of the categorical axis is it's not continuous -- it doesn't fill holes where there is no data. To handle this, I found the min and max weeks and made the x-axis always a continuous range between them. 

This does a good job of lining up the bars with the ticks, but more could still be done about overlapping labels. 

This is what a week-based chart (with gaps in the data) looked like before:
Screenshot 2024-05-02 at 15 56 46

And here's what it looks like with this change
Screenshot 2024-05-02 at 15 57 24

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts [hunk 1]
```diff
@@ -1115,6 +1149,10 @@ const getEchartAxes = ({
                         : showGridX,
                 },
                 inverse: !!xAxisConfiguration?.[0].inverse,
+                data: useWeekAxisConfiguration ? customXRange : undefined,
```
Comment (ZeRego) [inline packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts:1152]: We need to apply this to all necessary axis.
example when I flip axis

Screenshot 2024-05-03 at 10 05 27

File: packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts [hunk 2]
```diff
@@ -1115,6 +1149,10 @@ const getEchartAxes = ({
...
+                axisTick: useWeekAxisConfiguration
+                    ? { alignWithLabel: true, interval: 0 }
+                    : undefined,
```
Comment (ZeRego) [inline packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts:1155]: Curious about this change as ticks seem to be used as delimiters for categorical axis. So should we change this for all categorical axis or for none or just for weeks 🤔 
Example 

Screenshot 2024-05-03 at 10 07 49

File: packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts (hunk 1 above)
Comment (magnew) [reply packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts:1152]: Good call. I meant to do that.

File: packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts (hunk 2 above)
Comment (magnew) [reply packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts:1155]: I think just for weeks. We want weeks to behave more like a time axis, but we have to use category axis to get there. In the time axes, the ticks line up with the data and labels.

File: packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts (hunk 1 above)
Comment (magnew) [reply packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts]: Done

File: packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts [hunk 3]
```diff
@@ -931,6 +939,7 @@ const getEchartAxes = ({
             // This is to ensure the value is correctly formatted on some types
             switch (axisItem.timeInterval) {
                 case TimeFrames.WEEK_NUM:
+                case TimeFrames.WEEK:
                     axisConfig.axisLabel = {
                         formatter: (value: any) => {
                             return formatItemValue(axisItem, value, false);
```
[the remark this answers is not in the document]
Comment (magnew) [reply packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts:975]: Good question. No, these are already in UTC, converting them would result in moving the times TO utc, which isn't correct. We definitely have some unclarity in the code around this at the moment. I think it's getting better, but its not great that this isn't very clear.

**Review by ZeRego — APPROVED** [review]

**Pull request merged**